src/application/keyboard.rs

```diff
@@ -28,6 +29,7 @@ pub struct KeyboardInner {
     picker: RefCell<WeakRef<Picker>>,
     selected: Cell<Option<usize>>,
     stack: DerefCell<gtk::Stack>,
+    window: RefCell<WeakRef<MainWindow>>,
```

Comment:
It should be possible to use https://gtk-rs.org/docs/gtk/trait.WidgetExt.html#tymethod.get_parent_window, and downcast to the appropriate type.